cmd/tiller-proxy/main.go

```diff
@@ -173,6 +186,17 @@ func main() {
 		negroni.Wrap(handler.WithParams(h.DeleteRelease)),
 	))
 
+	// Chartsvc reverse proxy
+	chartsvc := r.PathPrefix("/chartsvc").Subrouter()
+	// Logos don't require authentication so bypass that step
+	chartsvc.Methods("GET").Path("/v1/assets/{repo}/{id}/logo").Handler(negroni.New(
+		negroni.Wrap(handler.WithoutParams(h.ProxyChartSVC)),
+	))
+	chartsvc.Methods("GET").Handler(negroni.New(
+		authGate,
+		negroni.Wrap(handler.WithoutParams(h.ProxyChartSVC)),
+	))
+
```

Comment:
So, after reading the docs and example for `gorilla/mux` I see that if we treated all traffic equally here, we can replace the above 10 lines with:
```
const chartsvcPrefix = "/chartsvc"
r.PathPrefix(chartsvcPrefix).Handler(http.StripPrefix(chartsvcPrefix, h.ChartsvcProxy)
```
and no longer need to define an actual handler, I think? Less code to test and maintain :)